src/aria/widgets/form/CheckBox.js

```diff
@@ -229,10 +243,12 @@ module.exports = Aria.classDefinition({
         _onBoundPropertyChange : function (propertyName, newValue, oldValue) {
             if (propertyName === 'value') {
                 this._cfg.value = newValue;
+                this._updateWAIAttributes('aria-checked', newValue);
                 this._setState();
                 this._updateDomForState();
             } else if (propertyName === 'disabled') {
                 this._cfg.disabled = newValue;
+                this._updateWAIAttributes('aria-disabled', newValue);
```

Comment:
For the sake of consistency, I would also move this line to `_updateDomForState`.